src/pytorch_lightning/accelerators/mps.py

```diff
@@ -24,7 +24,7 @@
 
 # For using the `MPSAccelerator`, user's machine should have `torch>=1.12`, Metal programming framework and
 # the ARM-based Apple Silicon processors.
-_MPS_AVAILABLE = _TORCH_GREATER_EQUAL_1_12 and torch.backends.mps.is_available() and platform.processor() == "arm"
+_MPS_AVAILABLE = _TORCH_GREATER_EQUAL_1_12 and torch.backends.mps.is_available() and platform.processor() == "arm64"
```

Comment:
@Birch-san On my M1 MAX MBP 2021, `platform.processor()` is `"arm"`, so the current patch excludes the case. It seems the function just uses `uname -p` (https://github.com/python/cpython/blob/89f52293281b6efc4ef666ef25e677683821f4b9/Lib/platform.py#L762), so the following should work in both cases.

```suggestion
_MPS_AVAILABLE = _TORCH_GREATER_EQUAL_1_12 and torch.backends.mps.is_available() and "arm" in platform.processor()
```

Reply:
It also returns "arm" on my machine (M1 Max MBP 2021 as well). Shouldn't the result then be consistent?

Reply:
Returns "arm" for me. Doesn't seem to matter which Python version, I tried 3.8, 3.9, 3.10. 
I would go with `platform.processor() in ("arm", "arm64")` for now.

Reply:
```suggestion
_MPS_AVAILABLE = _TORCH_GREATER_EQUAL_1_12 and torch.backends.mps.is_available() and platform.processor() in ("arm", "arm64")
```